client/llb/exec.go

```diff
@@ -153,7 +153,13 @@ func (e *ExecOp) Marshal(ctx context.Context, c *Constraints) (digest.Digest, []
 	}
 	if c.Caps != nil {
 		if err := c.Caps.Supports(pb.CapExecMetaSetsDefaultPath); err != nil {
-			env = env.SetDefault("PATH", system.DefaultPathEnv)
+			os := "linux"
```

Comment:
Shouldn't this be `runtime.GOOS`?

Reply:
No, this is client side and does not represent the exec os. Usually this should be set and unix is better default than windows. Note also that this is only executed on missing cap that was added a long time already so very theoretical to ever reach these lines.